sift/__main__.py

```diff
@@ -226,12 +226,202 @@ def _common_path_prefix(paths):
+        self.ui.animationSlider.valueChanged.connect(self.animation_slider_jump_frame)
+
+        # allow animation, once stopped, to propagate visibility to the document and layerlist:
+        self.scene_manager.didChangeLayerVisibility.connect(self.document.animation_changed_visibility)
+
+        self.document.didSwitchLayerSet.connect(self.animation_reset_by_layer_set_switch)
+        self.document.didChangeLayerVisibility.connect(self.update_frame_time_to_top_visible)
+        self.document.didReorderLayers.connect(self.update_frame_time_to_top_visible)
+        self.document.didRemoveLayers.connect(self.update_frame_time_to_top_visible)
+        self.document.didAddBasicLayer.connect(self.update_frame_time_to_top_visible)
+        self.document.didAddCompositeLayer.connect(self.update_frame_time_to_top_visible)
+
+    def next_frame(self, *args, **kwargs):
+        """Advance a frame along the animation order."""
+        self.scene_manager.layer_set.animating = False
+        self.scene_manager.layer_set.next_frame()
+
+    def prev_frame(self, *args, **kwargs):
+        """Retreat a frame along the animation list."""
+        self.scene_manager.layer_set.animating = False
+        self.scene_manager.layer_set.next_frame(frame_number=-1)
+
+    def reset_frame_slider(self, *args, **kwargs):
+        """Reset frame slider to show current animation state in document when aniamtion list changes."""
+        frame_count = len(self.document.current_animation_order)
+        frame_index = None  # self.scene_manager.layer_set._frame_number  # FIXME BAAD
+        self.ui.animationSlider.setRange(0, frame_count - 1)
+        self.ui.animationSlider.setValue(frame_index or 0)
+        self.ui.animPlayPause.setDown(False)
+        self.ui.animationSlider.repaint()
+        self.update_frame_time_to_top_visible()
+
+    def update_frame_slider(self, frame_info):
+        """
+        Update animation frame slider and time display to reflect current document animation state.
+        Args:
+            frame_info: tuple, ultimately from SceneGraphManager.layer_set callback
+        """
+        frame_index, frame_count, animating, uuid = frame_info[:4]
+        self.ui.animationSlider.setRange(0, frame_count - 1)
+        self.ui.animationSlider.setValue(frame_index or 0)
+        # LOG.debug('did update animation slider {} {}'.format(frame_index, frame_count))
+        self.ui.animPlayPause.setDown(animating)
+        self.ui.animationSlider.repaint()
+        if animating:
+            self.ui.animationLabel.setText(self.document.time_label_for_uuid(uuid))
+        else:
+            self.update_frame_time_to_top_visible()
+
+    def update_frame_time_to_top_visible(self, *args):
+        """Update frame slider's time display to show the current top layer's time."""
+        # FUTURE: don't address layer set directly
+        self.ui.animationLabel.setText(self.document.time_label_for_uuid(self.scene_manager.layer_set.top_layer_uuid()))
```

Comment:
Doesn't the doc have a method/property for what should be the current top layer?

Reply:
document.current_visible_layer_uuid may be appropriate if it's equivalent...

Reply:
I just approved this but realized this was never changed. Are we going to regret not changing this in the future?

Reply:
Just pushed a one-liner to use the preferred document property; quick testing shows things still working as expected.